dspace-spring-rest/src/main/java/org/dspace/app/rest/model/query/RestSearchOperator.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.model.query;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public enum RestSearchOperator {
```

Comment:
Needs javadocs.  Also could describe the regexs just below, as they are not all self-explanatory.